overlord/snapstate/handlers.go

```diff
@@ -3586,6 +3755,37 @@ func (m *SnapManager) doRefreshAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func checkSetupAliasesFailedInPruneMode(t *state.Task, snapName string) (bool, error) {
+	// check if we are in the undo of set-auto-aliases
+	if t.Kind() != "set-auto-aliases" || t.Status() != state.UndoingStatus {
+		return false, nil
+	}
+	// check if prunning was attempted
+	var pruneMode bool
+	if err := t.Get("prune-old-aliases", &pruneMode); err != nil && !errors.Is(err, state.ErrNoState) {
+		return false, err
+	}
+	if !pruneMode {
+		return false, nil
+	}
+	// check if setup-aliases ran after set-auto-aliases and failed
+	for _, tsk := range t.HaltTasks() {
+		if tsk.Kind() != "setup-aliases" || tsk.Status() != state.ErrorStatus {
+			continue
+		}
+		taskSetup, err := TaskSnapSetup(tsk)
+		if err != nil {
+			return false, err
+		}
+		if taskSetup.InstanceName() != snapName {
+			continue
+		}
+		// bingo
+		return true, nil
```

Comment:
if we look as snapst.AliasPending I'm not sure we need this bit here?

Reply:
this simplifies a lot, thank you!